docker/ubuntu14.04/Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM ubuntu:14.04
+ENV DEBIAN_FRONTEND noninteractive
+
+## Update cache and upgrade image
+RUN apt-get -y update && apt-get -y upgrade && apt-get -y dist-upgrade
+
+## Build environment packages
+RUN apt-get install -qq -y --ignore-missing \
+	apt-utils		\
+	automake		\
+        bc                      \
+	build-essential		\
+	bzip2			\
+	cmake			\
+	curl			\
+	git			\
+	libcurl4-openssl-dev	\
+	libssl-dev		\
+	make			\
+	pkg-config		\
+	python			\
+	sudo			\
+	tar			\
+	zip			\
+	unzip			\
+	wget			\
+	zlib1g-dev
+
+RUN mkdir -p /usr/local/bin
+
+## Install cmake since it's an expensive operation and best be done once
+COPY ./setup-cmake.sh /usr/local/bin/setup-cmake.sh
```

Comment:
This dramatically speeds up incremental build on old Ubuntu, as old distro only has CMake2 and the build of CMake-3.18 itself is very expensive operation. Ideally this could have been handled by providing CMake PPA for all distros. But we build from source (once only, per cached docker image).

Reply:
As these scripts only require to be run for one time, why copying them into /usr/local/bin, instead of running it in place?

Reply:
These still exist in prebuilt docker image. So, say, if you want to re-install or upgrade - you can reuse the script to pass a newer version of CMake to that and it'll upgrade the CMake version just for your current Docker session.